src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java

```diff
@@ -0,0 +1,94 @@
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SearchResult;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(MockitoExtension.class)
+class CcdApiSearchTest {
+
+    @Mock CoreCaseDataApi feignCcdApi;
+    @Mock CcdAuthenticatorFactory authenticatorFactory;
+    @Mock ServiceConfigProvider serviceConfigProvider;
+
+    @Mock CcdAuthenticator ccdAuthenticator;
+
+    ArgumentCaptor<String> idamTokenCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> s2sTokenCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> caseTypeCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> searchQueryCaptor = ArgumentCaptor.forClass(String.class);
+
+    CcdApi ccdApi;
+
+    @BeforeEach
+    void setUp() {
+        ccdApi = new CcdApi(feignCcdApi, authenticatorFactory, serviceConfigProvider);
+    }
+
+    @Test
+    void getCaseRefsByEnvelopeId_should_call_ccd_to_find_cases() {
+        // given
+        String envelopeId = "abc123";
+        String service = "hello";
+
+        String idamToken = "idam-token";
+        String s2sToken = "s2s-token";
+
+        ServiceConfigItem serviceCfg = cfg("some-jurisdiction", asList("a", "b"));
+
+        given(serviceConfigProvider.getConfig(service)).willReturn(serviceCfg);
+
+        given(authenticatorFactory.createForJurisdiction(serviceCfg.getJurisdiction())).willReturn(ccdAuthenticator);
+        given(ccdAuthenticator.getServiceToken()).willReturn(s2sToken);
+        given(ccdAuthenticator.getUserToken()).willReturn(idamToken);
+
+        given(
+            feignCcdApi.searchCases(
+                idamTokenCaptor.capture(),
+                s2sTokenCaptor.capture(),
+                caseTypeCaptor.capture(),
+                searchQueryCaptor.capture()
+            )
+        ).willReturn(
+            SearchResult
+                .builder()
+                .cases(asList(
+                    CaseDetails.builder().id(111L).build(),
+                    CaseDetails.builder().id(222L).build()
+                ))
+                .build()
+        );
+
+        // when
+        List<Long> caseRefs = ccdApi.getCaseRefsByEnvelopeId(envelopeId, service);
+
+        // then
+        assertThat(caseRefs).containsExactly(111L, 222L);
+
+        assertThat(idamTokenCaptor.getValue()).isEqualTo(idamToken);
+        assertThat(s2sTokenCaptor.getValue()).isEqualTo(s2sToken);
+        assertThat(caseTypeCaptor.getValue()).isEqualTo("a,b"); //concatenated case types for current service
+        assertThat(searchQueryCaptor.getValue()).contains(envelopeId);
+    }
+
+    private ServiceConfigItem cfg(String jurisdiction, List<String> caseTypes) {
```

Comment:
`cfg` is fine within the method, but maybe we should give the method itself a full name?

Reply:
fixed